examples/userapi/03_subdomains.ipynb

```diff
@@ -230,6 +232,10 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
+    "The Devito `SubDomain` API features two steps. The first of these is to create a subclass of `SubDomain`, which encapsulates the template by which `SubDomain`s should be constructed. This is not tied to a specific `Grid`. On a basic level, this specification is created by defining a suitable `define` method for the subclass.\n",
+    "\n",
+    "Note if one intends to introduce a custom `__init__` when subclassing `SubDomain`, then it is necessary for this `__init__` to take the `kwarg` `grid=None`, passing it through to `super().__init__(grid=grid)`. Such a custom `__init__` may be useful in applications such as creating many `SubDomain` instances with similar properties from a single template.\n",
```

Comment:
Is there an example of this somewhere?

Reply:
`userapi/04` uses it fairly extensively. as do various tests

Reply:
Sure